src/app/cli/src/init/client.ml

```diff
@@ -549,6 +549,75 @@ let send_payment_graphql =
          printf "Dispatched payment with ID %s\n"
            ((response#sendPayment)#payment |> unwrap_user_command)#id ))
 
+let sign_payment =
+  let open Command.Param in
+  let open Cli_lib.Arg_type in
+  let receiver_arg =
+    flag "--receiver" ~aliases:["receiver"]
+      ~doc:"PUBLICKEY Public key to which you want to send money"
+      (required public_key_compressed)
+  in
+  let amount_arg =
+    flag "--amount" ~aliases:["amount"]
+      ~doc:"VALUE Payment amount you want to send" (required txn_amount)
+  in
+  let token_arg =
+    flag "--token" ~aliases:["token"]
+      ~doc:"TOKEN_ID The ID of the token to transfer" (optional token_id)
+  in
+  let wallet_path_arg =
+    flag "--wallet-file" ~doc:"wallet filepath" (optional string) in
+  let secret_key_path_arg =
+    flag "--secret-key-file" ~doc:"secret key filepath" (optional string) in
+  let args =
+    Args.zip5 Cli_lib.Flag.signed_command_common receiver_arg amount_arg
+      token_arg wallet_path_arg secret_key_path_arg
+  in
+  Command.async ~summary:"generate a signature for a payment"
+    (Command.Param.map args
+       ~f:(fun (common, receiver_pk, amount, token_id, wallet_path, secret_key_path_arg) () ->
+            let {Cli_lib.Flag.sender; fee; nonce; memo} = common in
+            let fee_token = Token_id.default in
+            let fee_payer_pk = sender in
+            let valid_until = Unsigned.UInt32.max_int in
+            let nonce = Option.value ~default:Unsigned.UInt32.zero nonce in
+            let memo = Option.map ~f:Signed_command_memo.of_string memo
+                        |> Option.value ~default:Signed_command_memo.empty in
+            let common: Signed_command_payload.Common.t =
+              { fee; fee_token; fee_payer_pk; nonce; valid_until; memo } in
+            let body: Signed_command_payload.Body.t =
+              let token_id = Option.value ~default:Token_id.default token_id in
+              Signed_command_payload.Body.Payment
+              Payment_payload.Poly.
+              { source_pk=sender; receiver_pk; token_id; amount } in
+            let payload = Signed_command_payload.Poly.{ common; body } in
+            let logger = Logger.create () in
+            let%bind wallets =
```

Comment:
Can we avoid loading the directory when we don't need to? Perhaps something like
```ocaml
let%bind sk_file =
  match secret_key_path_arg with
  | Some x -> return x
  | None -> ...
```

Reply:
done!

Reply:
Can you move this let into the `None` branch? There's no reason for us to run this code if the user gave a path.

Reply:
```ocaml
let%bind sk_file =
  match secret_key_path_arg with
  | Some secret_key_path -> return secret_key_path
  | None ->
      let%bind wallets = ... in
      Secrets.Wallets.get_path wallets sender
in
```

Reply:
done!